p2p/host/routed/routed.go

```diff
@@ -135,10 +134,10 @@ func (rh *RoutedHost) Connect(ctx context.Context, pi peer.AddrInfo) error {
 			pi.Addrs = newAddrs
 			return rh.host.Connect(ctx, pi)
 		}
-
-		return err
+		// No appropriate new address found.
+		// Return the original dial error.
+		return cerr
```

Comment:
I'm just thinking but maybe it's worth returning an `errors.Join` of findPeerAddrs's error and cerr.
For now let's keep it as-is.